gnovm/pkg/gnolang/realm.go

```diff
@@ -496,11 +507,13 @@ func (rlm *Realm) processNewDeletedMarks(store Store) {
 			rlm.decRefDeletedDescendants(store, oo)
 		}
 	}
+	// clear deleted from all other sets.
+	rlm.clearDeleted()
```

Comment:
maybe this should be moved on `FinalizeRealmTransaction`

Reply:
I think each has its own advantages, as part of processNewDeleted also makes sense.